src/tasks/minions/camdozaalActivity/camdozaalFishingActivity.ts

```diff
@@ -0,0 +1,196 @@
+import { randInt } from 'e';
+import { Bank, LootTable } from 'oldschooljs';
+import { itemID } from 'oldschooljs/dist/util';
+
+import { Emoji, Events } from '../../../lib/constants';
+import addSkillingClueToLoot from '../../../lib/minions/functions/addSkillingClueToLoot';
+import Fishing from '../../../lib/skilling/skills/fishing';
+import { Fish, SkillsEnum } from '../../../lib/skilling/types';
+import { ActivityTaskOptionsWithQuantity } from '../../../lib/types/minions';
+import { roll, skillingPetDropRate } from '../../../lib/util';
+import { handleTripFinish } from '../../../lib/util/handleTripFinish';
+import { makeBankImage } from '../../../lib/util/makeBankImage';
+
+// Types of fish in camdozaal
+const camdozaalFishes: Fish[] = [
+	{
+		level: 7,
+		xp: 8,
+		id: itemID('Raw guppy'),
+		name: 'Raw guppy',
+		timePerFish: 1
+	},
+	{
+		level: 20,
+		xp: 16,
+		id: itemID('Raw cavefish'),
+		name: 'Raw cavefish',
+		timePerFish: 1
+	},
+	{
+		level: 33,
+		xp: 24,
+		id: itemID('Raw tetra'),
+		name: 'Raw tetra',
+		timePerFish: 1
+	},
+	{
+		level: 46,
+		xp: 33,
+		id: itemID('Raw catfish'),
+		name: 'Raw catfish',
+		timePerFish: 1
+	}
+];
+
+export const camdozaalFishingTask: MinionTask = {
+	type: 'CamdozaalFishing',
+	async run(data: ActivityTaskOptionsWithQuantity) {
+		let { userID, channelID, quantity, duration } = data;
+		const user = await mUserFetch(userID);
+		const currentFishLevel = user.skillLevel(SkillsEnum.Fishing);
+
+		// Fish types inside camdozaal
+		const guppy = camdozaalFishes.find(_fish => _fish.name === 'Raw guppy')!;
+		const cavefish = camdozaalFishes.find(_fish => _fish.name === 'Raw cavefish')!;
+		const tetra = camdozaalFishes.find(_fish => _fish.name === 'Raw tetra')!;
+		const catfish = camdozaalFishes.find(_fish => _fish.name === 'Raw catfish')!;
+
+		// Loot table based on users fishing level
+		const camdozaalFishTable = new LootTable();
+		if (currentFishLevel >= guppy.level) {
```

Comment:
Could reverse this table so that guppy is always added at the end of you are preventing them fishing in camdozaal without sufficient level

Reply:
The camdozaalFishTable already handles this